capellambse/_diagram_cache.py

```diff
@@ -0,0 +1,372 @@
+# SPDX-FileCopyrightText: Copyright DB Netz AG and the capellambse contributors
+# SPDX-License-Identifier: Apache-2.0
+"""Module to populate the diagram cache with native Capella diagrams."""
+
+from __future__ import annotations
+
+import datetime
+import json
+import logging
+import os
+import re
+import shlex
+import shutil
+import subprocess
+import tempfile
+import textwrap
+import typing as t
+from pathlib import Path
+
+from lxml import builder
+from lxml import etree as ET
+
+from capellambse.model.diagram import Diagram
+
+E = builder.ElementMaker()
+LOGGER = logging.getLogger(__name__)
+VALID_IMG_FMTS = (
+    "bmp",
+    "gif",
+    "jpg",
+    "png",
+    "svg",
+)
+
+
+class DiagramCacheCfg(t.NamedTuple):
+    capella_cli: str
+    image_format: t.Literal["bmp", "gif", "jpg", "png", "svg"]
+    create_index: bool
+    force: t.Literal["exe", "img"] | None
+    capella_project_name: str
+    diagrams_dir: Path
+    diagrams: list[Diagram]
+
+
+class DiagramCache:
+    _cli_cmd: list[str] = []
+    _cli_type: t.Literal["exe", "img"] | None = None
+    _index: list[dict[str, str | bool]] = []
+
+    def __init__(self, cfg: DiagramCacheCfg):
+        self.cfg = cfg
+        self._json_index_path = cfg.diagrams_dir / "index.json"
+        self._process_config()
+
+    def _find_exe(self, path) -> str | None:
+        def _is_exe(path: Path) -> bool:
+            return path.is_file() and os.access(path, os.X_OK)
+
+        fpath, _ = os.path.split(path)
+        if fpath:
+            if _is_exe(Path(path)):
+                return path
+        else:
+            for path_ in os.environ["PATH"].split(os.pathsep):
+                exe_file = Path(os.path.join(path_, path))
+                if _is_exe(exe_file):
+                    return str(exe_file)
+
+        return None
+
+    def _process_config(self) -> None:
+        if self.cfg.image_format.lower() not in VALID_IMG_FMTS:
+            raise ValueError(
+                f"Invalid image format '{self.cfg.image_format}'! "
+                "Image formats supported by the diagram cache: "
+                ", ".join(VALID_IMG_FMTS)
+            )
+        if self.cfg.force not in (
+            "exe",
+            "img",
+            None,
+        ):
```

Comment:
This is how `black` formats it.
Replaced tuple by a set, see efb8dde

Reply:
> This is how `black` formats it.

Weird, if I apply this patch locally and re-run black, it keeps it on one line. Are you using the correct version as per the pre-commit-config, i.e. v22.10? (Most importantly v22.X, because v23.X changes quite a few things)